flask_resources/serializers/json.py

```diff
@@ -72,6 +73,55 @@ def serialize_object_list(self, obj_list):
         return json.dumps(obj_list, cls=self.encoder, **self.dumps_options)
 
 
+class CustomJSONSerializer:
+    """JSON serializer that allows extra information to be dumped."""
+
+    def __init__(
+            self, object_key, object_schema_cls, list_schema_cls, serializer
+    ):
+        self.object_key = object_key
+        self.object_schema_cls = object_schema_cls
+        self.list_schema_cls = list_schema_cls
+        self.serializer = serializer
+
+    #
+    # Dump Python dictionary (obj and list)
+    #
+    def dump_obj(self, obj):
+        """Dump the object with extra information."""
+        ser_obj = deepcopy(obj)
+        ser_obj[self.object_key] = self.object_schema_cls().dump(ser_obj)
+
+        return ser_obj
+
+    def dump_list(self, obj_list):
+        """Dump the list of objects with extra information."""
+        ctx = {
+            "object_key": self.object_key,
```

Comment:
I am not sure about this `object_key`: it looks like very specific to the instance of `object_schema_cls` or list cls.
I think that you should pass a context instead.
In the constructor:
```
def __init__(
            self, serializer, object_schema_cls, list_schema_cls, schema_context={}
    ):
```
and then:
```
ctx = {
  "object_schema_cls": self.object_schema_cls,
}
ctx.update(self.schema_context)
self.[object_schema_cls | list_schema_cls](context=ctx)...
```

The `object_schema_cls` will have to change and accept a context. It should do the: 
```
ser_obj[self.object_key] = ...
```
inside the `dump` method, as the list obj is doing.